### PR TITLE
Avoid to reset html to empty string if block is heading and platform is android

### PR DESCRIPTION
**Pull request by marecar3**
This PR fixes the issue : [Heading block shows no formatting at first](https://github.com/wordpress-mobile/gutenberg-mobile/issues/627)

**Summary:**

PR for placeholder fix https://github.com/WordPress/gutenberg/pull/13699/
has introduced heading issue on Android https://github.com/wordpress-mobile/gutenberg-mobile/issues/627

**Note for testing:**

This fix will intentionally introduce original issue with a placeholder (on Android)
 which has lower priority then heading issue.

The new issue is raised: https://github.com/wordpress-mobile/gutenberg-mobile/issues/707

### CODE REVIEW

**Review by koke — COMMENTED** [review]
I understand what the code is trying to do but not why the bug was happening. What is different on iOS? Why is this not a problem on paragraph?

File: packages/editor/src/components/rich-text/index.native.js [hunk 1]
```diff
@@ -69,6 +70,10 @@ const gutenbergFormatNamesToAztec = {
 	'core/strikethrough': 'strikethrough',
 };
 
+const headingTags = [
+	'h1', 'h2', 'h3', 'h4', 'h5', 'h6',
```
Comment (hypest) [inline packages/editor/src/components/rich-text/index.native.js]: Adding headings related constants and logic seems out of place inside the RichText component, adding a "cyclic" like relationship between the RichText and the heading block.

What do you think instead, to add a flag as a prop to RichText which we will use to bypass the setting of the `html` variable to empty later in the render method?

**Comment by hypest** [comment]
Let me try to address your questions @koke :

> why the bug was happening

On Aztec-Android, Aztec is rendering text as heading only if the html passed has `<h>` tags. Empty text (like the one we pass if the block value is empty) ends up giving no "signal" to Aztec to turn on the heading rendering. But, if we send `<h2></h2>` to Aztec it will turn the rendering on and will correctly display any chars added after that fact as bold. But, the non-empty html will make the placeholder disappear... Aztec interprets the non empty html as non-empty content and will hide the placeholder. One could argue that this doesn't make much sense in visual mode but, that's a different discussion.

> Why is this not a problem on paragraph?

I'll assume you compare the case here with the Bold/Italic/Strikethrough formatting on the paragraph block. Those have a different implementation on Aztec (Android) and when toggling the format from the toolbar buttons, Aztec has some internal flags to turn the respective rendering ON and still treat the content as empty, thus showing the placeholder text.

Does that help with getting a better insight?

**Comment by koke** [comment]
I think it makes more sense now.

> Aztec has some internal flags to turn the respective rendering ON and still treat the content as empty, thus showing the placeholder text

Why doesn't the solution look more like this then? I imagine this is similar to what iOS calls `typingAttributes` and `RichText` calls `formatPlaceholder`. My thinking is that this is not specific to headings but we don't have any other content where this shows up.

File: packages/editor/src/components/rich-text/index.native.js (hunk 1 above)
Comment (marecar3) [reply packages/editor/src/components/rich-text/index.native.js]: ah yes! good point!

**Comment by koke** [comment]
I see that the aztec view on iOS is getting a `blockType` prop that contains the tag and [using that](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/react-native-aztec/ios/RNTAztecView/RCTAztecView.swift#L15) to figure out the styling for the placeholder, which is a different view laid on top of the empty aztec view

**Comment by marecar3** [comment]
@koke yes that is correct and on Android side, we are not using the `blockType` as in Aztec we are detecting the style by parsing the text with each user input. I tried to use on Android `blockType` but it can't help as it loses the placeholder

File: packages/editor/src/components/rich-text/index.native.js (hunk 1 above)
Comment (marecar3) [reply packages/editor/src/components/rich-text/index.native.js]: @hypest fixed, let me know if you have further comments.

**Comment by hypest** [comment]
> Why doesn't the solution look more like this then? I imagine this is similar to what iOS calls typingAttributes and RichText calls formatPlaceholder. My thinking is that this is not specific to headings but we don't have any other content where this shows up.

Can't be sure if an Aztec-level solution will be too hard or not right now. It would also increase the scope of the fix since my impression is that since Aztec-android is not supporting this behavior yet.

@koke , do you prefer if we postpone this fix until a new solution (deeper into Aztec) comes forth?

**Review by hypest — APPROVED** [review]
Verified this is working OK on Android and fixes the issue.

I'll 👍it but let's also [wait for Koke](https://github.com/WordPress/gutenberg/pull/14301#issuecomment-470644453) before finalizing.

**Comment by koke** [comment]
The latest changes are better and I think it makes sense to have this as a workaround for the next release. After merging this, rather than continuing down this path, I'd rather work on the placeholder logic on RNAztec to do something similar to iOS, and ensure that the placeholder also shows the same formatting as the content would.

**Comment by hypest** [comment]
Agreed @koke . The solution details can be worked out as part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/707.

Will merge this now.

**Pull request merged**